frontend/src/components/common/ContextMenu.js

```diff
@@ -0,0 +1,234 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import MetacodeSelect from '../MetacodeSelect'
+
+class ContextMenu extends Component {
+  static propTypes = {
+    topicId: PropTypes.string,
+    mapId: PropTypes.string,
+    currentUser: PropTypes.object,
+    map: PropTypes.object,
+    contextNode: PropTypes.object,
+    contextEdge: PropTypes.object,
+    contextPos: PropTypes.object,
+    contextFetchingSiblingsData: PropTypes.bool,
+    contextSiblingsData: PropTypes.object,
+    metacodeSets: PropTypes.array,
+    contextDelete: PropTypes.func,
+    contextRemove: PropTypes.func,
+    contextHide: PropTypes.func,
+    contextCenterOn: PropTypes.func,
+    contextPopoutTopic: PropTypes.func,
+    contextUpdatePermissions: PropTypes.func,
+    contextOnMetacodeSelect: PropTypes.func,
+    contextFetchSiblings: PropTypes.func,
+    contextPopulateSiblings: PropTypes.func
+  }
```

Comment:
you could throw in some isRequired if any are super duper required

Reply:
what did you think about not prefixing the names with "context" for all these props?

Reply:
I agree that in the future we should. The reason I am not right is because there is a weird/bad pattern we are using where many/most components get called with ALLL the props generated by ReactApp.getProps(). So having them namespaced is important. I could make each one explicit at the point where the ContextMenu gets rendered, but that'd be ugly ugly.